### DISCUSSION TITLE
Save copy or export to JPG, PNG, WEBP

### DISCUSSION THREAD

**Discussion by panathos** [tannerhelland/PhotoDemon#467] [General]
Save without loss is a good thing. 
But Photodemon lacking to save current work in different formats without change main file name.

**Comment by tannerhelland** [comment]
Hello @panathos !

`File > Save as...` (or the identical `Save as...` button in the left toolbar) allows you to save your current work to a new format without changing the current file name.  

That command may auto-suggest a new filename, e.g. "filename (2).jpg" when saving to the current folder, but you can absolutely delete the (2) and use the original filename if you want.

Would it be useful to add a new preference to disable this behavior?  (Specifically, the automatic appending of (2) when using `Save as` in the current folder?)

**Reply by panathos** [reply]
Thank you @tannerhelland but my PhotoDemon 9.0 works differently: doing `Save as...`, file format change at every save. 

So if i start with "file.pdi" (or another format), doing `Save as...`  to PNG output "file (1).png" and change current filename to `file (1).png`. 
From here,  the `Save` command updates "file (1).png" and not original "file.pdi".

If there was a command to save versioned jobs, without changing the starting filename, that would be great for me.

**Comment by tannerhelland** [comment]
The latest nightly build (9.2 build 90) now provides a setting to toggle the way `Save As` suggests filenames.  Use the `Tools > Options > Saving` panel to switch behaviors.

![image](https://user-images.githubusercontent.com/1930029/209408329-a38084a7-0913-4348-bad9-8c9413e94012.png)

**Comment by ortegafernando** [comment]
Hi, I will like also to have an option in EXPORT menu (at least JPG and PNG). Why? Because I usually do some work with PDI file, and I want to export finally it to jpg or png, then I want to save PDI with other name, go on working, and save a second jpg/png file.

The problem here is that as we dont have any export option to jpg or png the step that I do now are:
1. save as jpg
2. re open again the original pdi
3. close the saved jpg
4. re name the pdi
5. go on working.

If we could have an option to export to JPG or PNG it will be like:
1. export to jpg
2. re name the pdi
3. go on working.

Do you understand me?
I think it is a more natural way to work, isnt it?

**Reply by tannerhelland** [reply]
Hello @ortegafernando .  Thank you for this comment.  I am the developer of PhotoDemon and I would like to understand your workflow better.

After saving an image as a JPEG, why do you need to reopen again the original PDI?

When you save an image as a JPEG, the image itself is not modified in any way (layers are not flattened, or compressed, or any changes made).  Instead, a temporary image copy is created on-the-fly, and that temporary copy is used to produce the JPEG file.  

The image itself, as you are working with it in PhotoDemon, is *never* modified by a "Save" action.

PhotoDemon also provides the "Save a lossless copy" button right there on the main toolbar (Shortcut `Ctrl+Alt+Shift S`) for quickly producing a lossless PDI file of the current image state.  This allows for constant lossless save operations as you work, 

I can look at adding a new `Export to image file` option, but as far as I can tell, this would simply work exactly like the `Save` option already does, with the sole exception of *not* updating the current working filename for the image.  Would that still be useful?

**Reply by ortegafernando** [reply]
Hi, sorry, I have not recieved any notification from you. I usually make little changes from the same PDI, so every time I do these changes, I need to make all steps.

May be you want to make a PDI file with multiple layers and by hidding or showing layers "export" several different jpg files. 

Export will be like an option to save as jpg but continue working in PDI file. 

May be this is clear. What do you think? Thanks.

**Reply by tannerhelland** [reply]
Thank you very much for the follow-up information, @ortegafernando .  The latest PhotoDemon nightly builds now include a dedicated `File > Export image to file` menu, and it has the same keyboard shortcut as Photoshop (`Alt+Shift+Ctrl+W`).  This lets you export to any format without affecting the current image's format or save state.

I hope this helps!

**Comment by hi5** [comment]
@panathos 
Not entirely sure I understand what you mean, but the way I see it you can already do that:

1. You have a PDI with multiple layers, text, etc etc. **Save** just to be sure: <kbd>Ctrl</kbd>+<kbd>S</kbd>.
2. Just **Save As** - <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>S</kbd>, enter `name you want.png` or `name you want.jpg`, the file is now saved as such.
3. now your file "looks" like it is `name you want.png` but you should notice it still has all the layers, text etc, so just so another **Save As**, this time as `YourNewName.PDI`.
4. Repeat step 1-3

And that is about it. Using an `Export As` would be a bit more work as these don't have shortcuts (I would assume).

**Comment by tannerhelland** [comment]
Thank you to everyone for their input on this topic.  The latest PhotoDemon nightly builds now include a much-improved `File > Export` menu.  There is now a dedicated `Export image to file` menu alongside some other new Export additions.

FYI - the new `Export image` menu uses the same keyboard shortcut as Photoshop (`Alt+Shift+Ctrl+W`) for users accustomed to that workflow.

I am always open to further improvements, and I hope the new Export menu items are a meaningful workflow improvement for those who need it.  Thanks again for everyone's comments on this feature.
